src/NetworkTest/testQuality/helpers/getPublisherRtcStatsReport.ts

```diff
@@ -0,0 +1,21 @@
+import { OT } from '../../types/opentok';
+
+export default (publisher: OT.Publisher): Promise<OT.PublisherRtcStatsReportArr | null> =>
+  new Promise((resolve) => {
+    // If getRtcStatsReport is not a function, that means OT version is < 2.17.6
+    // In this case we just return null.
+    if (typeof publisher.getRtcStatsReport !== 'function') {
+      return resolve(null);
+    }
+
+    // Need to evaluate the result of getRtcStatsReport() to determine
+    // whether it's the promise or callback version of the API.
+    const getRtcStatsReport = publisher.getRtcStatsReport();
+    if (!getRtcStatsReport) {
+      publisher.getRtcStatsReport((err, stats) => { resolve(stats); });
```

Comment:
Should we handle the error condition?

```suggestion
publisher.getRtcStatsReport((err, stats) => {
  if (stats) {
    resolve(stats);
  } else {
    resolve(null);
  }
});
```

Reply:
I think it's not needed for our use case, since in case of an error, we will return the stats that is going to be null.